src/components/Task/Drawer/Form/Form.tsx

```
@@ -191,6 +218,49 @@ const TaskDrawerForm = ({
     }
   };
 
+  const onDeleteTask = async (): Promise<void> => {
+    try {
+      if (task) {
+        await deleteTask({
+          variables: {
+            accountListId,
+            id: task.id,
+          },
+          update: (cache) => {
+            const query = {
+              query: GetTasksForTaskListDocument,
+              variables: {
+                accountListId,
+                first: rowsPerPage,
+                ...filter,
+              },
+            };
+            const dataFromCache = cache.readQuery<GetTasksForTaskListQuery>(
+              query,
+            );
+
+            cache.writeQuery({
+              ...query,
+              data: {
+                tasks: {
+                  ...dataFromCache.tasks,
+                  nodes: dataFromCache.tasks.nodes.filter(
+                    ({ id }) => id !== task.id,
+                  ),
+                },
+              },
+            });
+          },
+        });
+        enqueueSnackbar(t('Task deleted successfully'), { variant: 'success' });
+        handleRemoveDialog(false);
+        onClose();
+      }
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
```

If we want this error in Rollbar, we should probably rethrow the error.